storage/innobase/dict/dict0stats.cc

```diff
@@ -1372,6 +1406,91 @@ static dberr_t btr_pcur_open_level(btr_pcur_t *pcur, ulint level, mtr_t *mtr,
 }
 
 
+/** Statistics collected for a single B-tree level
+during index analysis. Used to track distinct key values,
+record counts, and page information when analyzing index
+levels for the query optimizer's statistics. */
+struct IndexLevelStats
+{
+  /** Level number */
+  ulint level;
+  /** Whether nulls are considered unequal for statistics */
+  bool nulls_unequal;
+  /** Array for number of distinct keys */
+  uint64_t* n_diff;
+  /** Array of number of non-null keys for all prefixes */
+  uint64_t* n_non_null;
+  /** Boundaries of groups of distinct keys */
+  boundaries_t* bounds;
+  /** Total number of records */
+  uint64_t n_recs;
+  /** Total number of pages */
+  uint32_t n_pages;
+
+  IndexLevelStats(ulint level_, unsigned stats_method,
+                  dict_index_t* index) : level(level_)
+  {
+    nulls_unequal= (stats_method > SRV_STATS_NULLS_EQUAL);
+    bool calc_non_null= (stats_method == SRV_STATS_NULLS_IGNORED);
+    if (level == 0)
+    {
+      n_diff= index->stat_n_diff_key_vals;
+      memset(n_diff, 0x0, index->n_uniq * sizeof(uint64_t));
```

Comment:
Instead of invoking `sizeof` on a type name, it would be better to invoke it on the variable name (`sizeof *n_diff`). In that way, there won’t be any inconsistency in the event that the data type is being changed.

This is not the only such occurrence; please check all of them.